spec/api-power-monitor-spec.js

```diff
@@ -0,0 +1,89 @@
+const Promise = require('bluebird')
+
+// lazy require powerMonitor so we can setup responses for dbus method calls
+function loadPowerMonitor () {
+  return require('electron').remote.powerMonitor
+}
+
+function powerMonitorTests () {
+  let logindMock, powerMonitor, getCalls, emitSignal, reset
+
+  before(async () => {
+    const systemBus = dbus.systemBus()
+    const loginService = systemBus.getService('org.freedesktop.login1')
+    const getInterface = Promise.promisify(loginService.getInterface, {context: loginService})
+    logindMock = await getInterface('/org/freedesktop/login1', 'org.freedesktop.DBus.Mock')
+    getCalls = Promise.promisify(logindMock.GetCalls, {context: logindMock})
+    emitSignal = Promise.promisify(logindMock.EmitSignal, {context: logindMock})
+    reset = Promise.promisify(logindMock.Reset, {context: logindMock})
+  })
+
+  after(async () => {
+    await reset()
+  })
+
+  describe('when powerMonitor module is loaded', () => {
+    function onceMethodCalled (done) {
+      function cb () {
+        logindMock.removeListener('MethodCalled', cb)
+      }
+      done()
+      return cb
+    }
+
+    before((done) => {
+      logindMock.on('MethodCalled', onceMethodCalled(done))
+      powerMonitor = loadPowerMonitor()
+    })
+
+    it('should call Inhibit to delay suspend', async () => {
+      const calls = await getCalls()
+      assert.equal(calls.length, 1)
+      assert.deepEqual(calls[0].slice(1), [
+        'Inhibit', [
+          [[{type: 's', child: []}], ['sleep']],
+          [[{type: 's', child: []}], ['electron']],
+          [[{type: 's', child: []}], ['Application cleanup before suspend']],
+          [[{type: 's', child: []}], ['delay']]
+        ]
+      ])
+    })
+
+    describe('when PrepareForSleep(true) signal is sent by logind', () => {
+      it('should emit "suspend" event', (done) => {
+        powerMonitor.once('suspend', () => done())
+        emitSignal('org.freedesktop.login1.Manager', 'PrepareForSleep',
+          'b', [['b', true]])
+      })
+
+      describe('when PrepareForSleep(false) signal is sent by logind', () => {
+        it('should emit "resume" event', (done) => {
+          powerMonitor.once('resume', () => done())
+          emitSignal('org.freedesktop.login1.Manager', 'PrepareForSleep',
+            'b', [['b', false]])
+        })
+
+        it('should have called Inhibit again', async () => {
+          const calls = await getCalls()
+          assert.equal(calls.length, 2)
+          assert.deepEqual(calls[1].slice(1), [
+            'Inhibit', [
+              [[{type: 's', child: []}], ['sleep']],
+              [[{type: 's', child: []}], ['electron']],
+              [[{type: 's', child: []}], ['Application cleanup before suspend']],
+              [[{type: 's', child: []}], ['delay']]
+            ]
+          ])
+        })
+      })
+    })
+  })
+}
+
+if (process.platform !== 'linux' || !process.env.DBUS_SYSTEM_BUS_ADDRESS) {
```

Comment:
I guess it's easier to make this check in the topmost `before` hook:

```js
describe('powerMonitor', () => {
  before(async function () {
    if (<...>) {
      this.skip()
      return
    }

    <...>
  })
})
```

Reply:
:+1: